ecommers/lib/core/cache/cache_database.dart

```diff
@@ -0,0 +1,54 @@
+import 'dart:async';
+
+import 'package:path_provider/path_provider.dart';
+import 'package:sembast/sembast.dart';
+import 'package:sembast/sembast_io.dart';
+import 'package:path/path.dart';
+
+class CacheDatabase {
+  Completer<Database> _dbCompleter;
```

Comment:
Completer is redundant here, you can register instance async in dependency service
 https://pub.dev/packages/get_it#synchronizing-asynchronous-initialisations-of-singletons